starboard/nplb/posix_compliance/posix_socket_helpers.cc

```diff
@@ -182,6 +185,76 @@ int PosixGetLocalAddressIPv6(sockaddr* address_ptr) {
   return result;
 }
 
+int port_number_for_tests = 0;
+pthread_once_t valid_port_once_control = PTHREAD_ONCE_INIT;
+
+void PosixInitializePortNumberForTests() {
+  // Create a listening socket. Let the system choose a port for us.
+  int socket_fd = PosixCreateBoundListeningTcpSocket();
+  SB_DCHECK(socket_fd >= 0);
+
+  // Query which port this socket was bound to and save it to valid_port_number.
+  SbSocketAddress socket_address = {0};
+
+  sbposix::SockAddr sock_addr;
+  int result_local =
+      getsockname(socket_fd, sock_addr.sockaddr(), &sock_addr.length);
+  SB_DCHECK(result_local >= 0);
+
+  if (!sock_addr.ToSbSocketAddress(&socket_address)) {
+    result_local = -1;
+  }
+
+  SB_DCHECK(result_local >= 0);
+  port_number_for_tests = socket_address.port;
+
+  // Clean up the socket.
+  bool result = HANDLE_EINTR(close(socket_fd)) >= 0;
+  SB_DCHECK(result);
+}
+
+int PosixGetPortNumberForTests() {
+  pthread_once(&valid_port_once_control, &PosixInitializePortNumberForTests);
+  return port_number_for_tests;
+}
+
+int PosixCreateBoundListeningTcpSocket() {
+  int socket_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_UDP);
+  if (socket_fd < 0) {
+    ADD_FAILURE() << "SbSocketCreate failed";
+    return -1;
+  }
+
+  int on = 1;
+  if (setsockopt(socket_fd, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on)) != 0) {
+    ADD_FAILURE() << "SbSocketSetReuseAddress failed";
+    HANDLE_EINTR(close(socket_fd));
+    // TODO: what return address do we want if this fails
+    return -1;
+  }
+
+  // TODO: posixify socket bind
+  SbSocketAddress address = GetUnspecifiedAddress(kSbSocketAddressTypeIpv4, 0);
+  int bind_result = SbSocketBind(socket_fd, &address);
```

Comment:
@maxz-lab - could you help me add the posix version of bind here? the rest of it *seemed* straightforward, but please correct me if anything else is wrong!

Reply:
Figured it out.